kieker-monitoring/src/kieker/monitoring/writer/serializer/AbstractContainerFormatSerializer.java

```diff
@@ -0,0 +1,86 @@
+/***************************************************************************
+ * Copyright 2015 Kieker Project (http://kieker-monitoring.net)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+package kieker.monitoring.writer.serializer;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+
+import kieker.common.configuration.Configuration;
+import kieker.common.record.IMonitoringRecord;
+import kieker.common.util.dataformat.FormatIdentifier;
+
+/**
+ * Abstract serializer for the Kieker container format for monitoring records. This container
+ * format provides a flexible mechanism to embed data in different formats.
+ *
+ * @author Holger Knoche
+ *
+ * @since 1.13
+ */
+public abstract class AbstractContainerFormatSerializer extends AbstractMonitoringRecordSerializer {
+
+	/** Container format identifier. */
+	public static final int CONTAINER_IDENTIFIER = FormatIdentifier.CONTAINER_FORMAT.getIdentifierValue();
+
+	private static final int HEADER_SIZE = 8;
+
+	/**
+	 * Creates a new serializer using the given configuration.
+	 *
+	 * @param configuration
+	 *            The configuration to use
+	 */
+	public AbstractContainerFormatSerializer(final Configuration configuration) {
+		super(configuration);
+	}
+
+	@Override
+	public final int serializeRecord(final IMonitoringRecord record, final ByteBuffer buffer) {
```

Comment:
If this class does not support the required method, then the abstraction `AbstractMonitoringRecordSerializer` does not seem to be chosen well. Perhaps, you should just provide `serializeRecords(..)` as isolated (utility) class.